src/odemis/acq/stitching/_registrar.py

```diff
@@ -614,52 +603,57 @@ def _get_shift(self, prev_tile, tile):
-            t2, b2 = -int(exp_shift[1]), tile.shape[0]
+        if exp_tile_dist_px[1] < 0:
+            t1, b1 = 0, tile.shape[0] + int(exp_tile_dist_px[1])
+            t2, b2 = -int(exp_tile_dist_px[1]), tile.shape[0]
         else:
-            t1, b1 = int(exp_shift[1]), tile.shape[0]
-            t2, b2 = 0, tile.shape[0] - int(exp_shift[1])
+            t1, b1 = int(exp_tile_dist_px[1]), tile.shape[0]
+            t2, b2 = 0, tile.shape[0] - int(exp_tile_dist_px[1])
+
+        # TODO should we take a larger area?
         prev_tile_roi = numpy.array(prev_tile)[t1:b1, l1:r1]
         tile_roi = numpy.array(tile)[t2:b2, l2:r2]
 
         # If you need to crop the tile without changing the output shift,
         # you can do it here with the pattern tile_roi[t:-b, l:-r]
-        shift = MeasureShift(tile_roi, prev_tile_roi)
-        shift_total = numpy.subtract(exp_shift, shift)
+        meas_tile_dist_px = MeasureShift(tile_roi, prev_tile_roi)
+        # How much to shift the tile relative to the metadata position
+        shift_px = numpy.subtract(exp_tile_dist_px, meas_tile_dist_px)
 
         # Measure accuracy (ncc value)
         avg = numpy.average(prev_tile), numpy.average(tile)
-        dist = prev_tile_roi - avg[0], tile_roi - avg[1]
-        covar = numpy.sum(dist[0] * dist[1]) / prev_tile_roi.size
-        var = numpy.sum(dist[0] ** 2) / prev_tile_roi.size, numpy.sum(dist[1] ** 2) / tile_roi.size
-        stDev = (numpy.sqrt(var[0]), numpy.sqrt(var[1]))
-        if stDev[0] == 0 or stDev[1] == 0:
-            return exp_shift, 0
+        diff = prev_tile_roi - avg[0], tile_roi - avg[1]
+        covar = numpy.sum(diff[0] * diff[1]) / prev_tile_roi.size
+        var = numpy.sum(diff[0] ** 2) / prev_tile_roi.size, numpy.sum(diff[1] ** 2) / tile_roi.size
+        st_dev = (numpy.sqrt(var[0]), numpy.sqrt(var[1]))
+        if st_dev[0] == 0 or st_dev[1] == 0:
+            return exp_tile_dist_px, 0
         # Normalized cross correlation (values between -1 and 1)
-        ncc = (covar / (stDev[0] * stDev[1]))
-        overlap = numpy.abs(numpy.subtract(tile.shape[::-1], numpy.abs(exp_shift)))  # tile.shape is YX, so reverse it
-        if numpy.any(numpy.abs(shift) > overlap):
-            logging.info(f"Calculated shift {shift} is larger than the overlap size {overlap}, "
+        ncc = (covar / (st_dev[0] * st_dev[1]))
+        overlap = numpy.abs(
+            numpy.subtract(tile.shape[::-1], numpy.abs(exp_tile_dist_px)))  # tile.shape is YX, so reverse it
+        if numpy.any(numpy.abs(meas_tile_dist_px) > overlap):
+            logging.info(f"Calculated shift {meas_tile_dist_px} is larger than the overlap size {overlap}, "
                          f"using expected position instead.")
-            return exp_shift, 0
+            return exp_tile_dist_px, 0
 
-        return shift_total, ncc
+        return shift_px, ncc
 
-    def _compute_registration(self, row, col):
+    def _compute_registration(self, tile, row, col):
```

Comment:
Is this param used?

Reply:
I think the idea is that this method has the same signature as the parent class.

Reply:
that's indeed the idea. But this also allows me to remove line 666, which I will do.